image/store.go

```diff
@@ -77,7 +73,7 @@ func (is *store) restore() error {
 		}
 		var l layer.Layer
 		if chainID := img.RootFS.ChainID(); chainID != "" {
-			l, err = is.ls.Get(chainID)
+			l, err = is.lss[img.OperatingSystem()].Get(chainID)
```

Comment:
In many places like these, the `os` does not seem to be validated and can easily cause a panic.

Reply:
Again, I will double check if there are unbounded checks.

Reply:
Have added further explicit checks. Updating coming shortly.

Reply:
The checks are now in, in the latest push. Pretty much anywhere across the code base before it accesses the layerstore by index, it does an explicit check to validate that a panic won't occur. This is probably very paranoid defensive coding and many are theoretically impossible to hit, but hopefully it gives some more reassurance.